admin_guide/high_availability.adoc

```diff
@@ -0,0 +1,132 @@
+= High Availability
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+This document describes how to make a network service highly available using virtual IP addresses and IP failover. Even though we specifically describe an example of how to setup a highly available router environment, it can be generically applied to any IP based service.
+
+== Why IP Failover?
+Within a cluster, the Kubernetes replication controller ensures that the
+deployment requirements - in particular the number of replicas - are satisfied when the appropriate resources are available.
+
+The link:../architecture/core_objects/routing.html[Router] is one such deployment which can be run with 2 or more replicas to provide a highly available service resilient to individual router instance failures. Depending on how these router instances are discovered (for example via a well-published service or a DNS entry or even IP addresses), this could well impose operational requirements to handle failure cases when 1 or more router instances are "unreachable".
+
+For some IP based traffic services (our example being the router), if we can ensure that the IP addresses are always serviced as long as a single instance is available, this simplifies the operational overhead and handles failure cases gracefully. It is pertinent to note here, that even though the service is highly available, performance could still be impacted.
+
+== Use Cases
+  1. As an administrator, I want my cluster to be assigned a resource set
+     and I want the cluster to automatically manage those resources.
+  2. As an administrator, I want my cluster to be assigned a set of virtual
+     IP addresses that the cluster manages and migrates (with zero or
+     minimal downtime) on failure conditions.
+  3. As an addendum to use case #2, the administrator should not be
+     required to perform any manual interaction to update the upstream
+     "discovery" sources (e.g. DNS). The cluster should service all the
+     assigned virtual IPs when atleast a single node is available - and
+     this should be inspite of the fact that the current available
+     resources are not sufficient to reach "critical mass" aka the
+     desired state.
+
+== ipfailover admin command
+The `oscadmin ipfailover` command helps to setup the VIP failover configuration. An administrator can configure IP failover on an entire cluster or as would normally be the case on a subset of nodes (as defined via a labelled selector). If you are running in production, it is recommended that the labelled selector for the nodes matches atleast 2 nodes to ensure you have failover protection and that you provide a --replicas=<n> value that matches the number of nodes for the given labelled selector.
```

Comment:
Is `oscadmin` mentioned here (and `oscadm` mentioned throughout the rest of the doc) just a typo of `osadm`? Or is the name for the admin CLI intending to change?